src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs

```diff
@@ -403,6 +409,41 @@ FROM attributes
         }
     }
 
+    public async Task InsertWalletAttribute(Guid walletId, WalletAttribute walletAttribute)
+    {
+        await _connection.ExecuteAsync(
+               @"INSERT INTO wallet_attributes(id, wallet_id, certificate_id, registry_name, attribute_key, attribute_value, salt)
+                  VALUES (@id, @walletId, @certificateId, @registryName, @attributeKey, @attributeValue, @salt)",
+               new
+               {
+                   id = Guid.NewGuid(),
+                   walletId,
+                   walletAttribute.CertificateId,
+                   walletAttribute.RegistryName,
+                   attributeKey = walletAttribute.Key,
+                   attributeValue = walletAttribute.Value,
+                   walletAttribute.Salt
+               });
+    }
+
+    public async Task<IEnumerable<WalletAttribute>> TryGetWalletAttributes(Guid walletId, Guid certificateId, string registryName, IEnumerable<string> keys)
```

Comment:
This method does not follow the Try-pattern. Should it just be renamed to `GetWalletAttributes`